src/nodes/Preview.js

```diff
@@ -84,13 +85,32 @@ export default Node.create({
 			 *
 			 */
 			unsetPreview: () => ({ state, chain }) => {
-				console.info(this.attributes)
 				return isActive(this.name, this.attributes, state)
 					&& chain()
 						.setNode('paragraph')
 						.run()
 			},
 
+			/**
+			 * Insert a preview for given link.
+			 *
+			 */
+			insertPreview: (link) => ({ state, chain }) => {
+				return chain()
+					.insertContent({
+						type: 'preview',
+						attrs: { href: link, title: 'preview' },
+						content: [{
+							type: 'text',
+							marks: [{
```

Comment:
Seems a bit unexpected that the preview node still has an additional mark, can you tell why?

Reply:
I think it matches the data structure created otherwise. We basically change the wrapping block node between paragraph and preview when switching between the two:

The `setPreview` function above converts the surrounding paragraph of a link into a preview node:
![grafik](https://github.com/nextcloud/text/assets/97337118/08e0b1c6-3bc9-4914-8fae-1ed2350c837f)

If you look at `unsetPreview` above it also only unwraps the link inside the preview node:
![grafik](https://github.com/nextcloud/text/assets/97337118/fa161e98-1cb3-4f31-bc1b-51ef53d699a8)